bin/varnishd/mgt/mgt_child.c

```diff
@@ -452,6 +453,21 @@ mgt_launch_child(struct cli *cli)
  * Cleanup when child dies.
  */
 
+static int
+kill_child(void) {
+	int i, error;
+
+	VJ_master(JAIL_MASTER_KILL);
+	if (MGT_FEATURE(FEATURE_NO_COREDUMP))
+		i = kill(child_pid, SIGKILL);
+	else
+		i = kill(child_pid, SIGQUIT);
+	error = errno;
+	VJ_master(JAIL_MASTER_LOW);
+	errno = error;
+	return (i);
```

Comment:
This is documented in #2010 and diagnosed (with verification) by martin. Through the jail system the mgt process decreases its rights, and becomes unable to kill the child process. That is why we need to elevate temporarily in this patch.